compose/service.py

```diff
@@ -395,27 +395,57 @@ def _containers_have_diverged(self, containers):
     def execute_convergence_plan(self,
                                  plan,
                                  do_build=True,
-                                 timeout=DEFAULT_TIMEOUT):
+                                 timeout=DEFAULT_TIMEOUT,
+                                 detached=False):
         (action, containers) = plan
 
         if action == 'create':
             container = self.create_container(do_build=do_build)
+
+            if container.has_api_logs and not detached:
+                container.attach_log_stream()
```

Comment:
`attach_log_stream()` already checks for `has_api_logs()` (which I think is a good place to do it), so I think we can remove it from here and do just `if not detached:`